cachi2/core/package_managers/yarn_classic/utils.py

```diff
@@ -0,0 +1,101 @@
+from collections import deque
+from typing import Any
+
+from pyarn.lockfile import Package as PYarnPackage
+
+from cachi2.core.package_managers.yarn_classic.project import PackageJson, YarnLock
+from cachi2.core.package_managers.yarn_classic.workspaces import Workspace
+from cachi2.core.rooted_path import RootedPath
+
+
+def find_non_dev_deps(
+    main_package_json: PackageJson,
+    yarn_lock: YarnLock,
+    workspaces: list[Workspace],
+) -> set[str]:
+    """
+    Identify all non-dev dependencies in the root package and its workspaces.
```

Comment:
Not sure, are `peerDependencies` and `optionalDependencies` considered runtime ?

Reply:
I think yes, as per [the docs](https://classic.yarnpkg.com/lang/en/docs/dependency-types/):
"""
peerDependencies
Peer dependencies are a special type of dependency that would only ever come up if you were publishing your own package.

Having a peer dependency means that your package needs a dependency that is the same exact dependency as the person installing your package. This is useful for packages like react that need to have a single copy of react-dom that is also used by the person installing it.
"""
and
"""
optionalDependencies
Optional dependencies are just that: optional. If they fail to install, Yarn will still say the install process was successful.

This is useful for dependencies that won’t necessarily work on every machine and you have a fallback plan in case they are not installed (e.g. Watchman).
"""

I read first as a fancy way to limit run time dependencies in a special case, and the second as a way to specify an _optional_ run time dependency.